adler32_fold.c

```diff
@@ -7,14 +7,18 @@
 #include "functable.h"
 #include "adler32_fold.h"
 
+#include <limits.h>
+
 Z_INTERNAL uint32_t adler32_fold_copy_c(uint32_t adler, uint8_t *dst, const uint8_t *src, uint64_t len) {
     adler = functable.adler32(adler, src, len);
-    while (len > SIZE_MAX) {
-        memcpy(dst, src, SIZE_MAX);
-        dst += SIZE_MAX;
-        src += SIZE_MAX;
-        len -= SIZE_MAX;
+#if INTPTR_MAX > SSIZE_MAX
```

Comment:
I think this could benefit from a one-liner comment about what this is a workaround for.

Reply:
Added explanation.